src/DonorProfiles/Tabs/DonationHistoryTab/DonationsRoute.php

```diff
@@ -43,11 +44,39 @@ public function handleRequest( $request ) {
 	 * @since 2.10.0
 	 */
 	protected function getData( DonationsRepository $repository, $donorId ) {
-		return [
-			'donations' => $repository->getDonations( $donorId ),
-			'count'     => $repository->getDonationCount( $donorId ),
-			'revenue'   => $repository->getRevenue( $donorId ),
-			'average'   => $repository->getAverageRevenue( $donorId ),
-		];
+
+		$donations = $repository->getDonations( $donorId );
+		$count     = $repository->getDonationCount( $donorId );
+		$revenue   = $repository->getRevenue( $donorId );
+		$average   = $repository->getAverageRevenue( $donorId );
+
+		if ( $donations && $count && $revenue && $average ) {
+
+			return new WP_REST_Response(
+				[
+					'status'        => 200,
+					'response'      => 'success',
+					'body_response' => [
+						[
+							'donations' => $donations,
+							'count'     => $count,
+							'revenue'   => $revenue,
+							'average'   => $average,
+						],
+					],
+				]
+			);
+		}
+
+		return new WP_REST_Response(
+			[
+				'status'        => 400,
+				'response'      => 'database_error',
+				'body_response' => [
+					'message' => __( 'An error occured while retrieving your donation records.', 'give' ),
```

Comment:
I resolved typo in different branch https://github.com/impress-org/givewp/compare/e3d998a1e132%5E...19ac117ffc1f